modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaUpsertCPDefinitionMethodCheck.java

```diff
@@ -0,0 +1,121 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.source.formatter.check.util.JavaSourceUtil;
+import com.liferay.source.formatter.parser.JavaClass;
+import com.liferay.source.formatter.parser.JavaClassParser;
+import com.liferay.source.formatter.parser.JavaMethod;
+import com.liferay.source.formatter.parser.JavaTerm;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Tamyris Bernardo
+ */
+public class UpgradeJavaUpsertCPDefinitionMethodCheck extends BaseUpgradeCheck {
+
+	@Override
+	protected String format(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		String newContent = content;
+
+		JavaClass javaClass = JavaClassParser.parseJavaClass(fileName, content);
+
+		for (JavaTerm childJavaTerm : javaClass.getChildJavaTerms()) {
+			if (!childJavaTerm.isJavaMethod()) {
+				continue;
+			}
+
+			JavaMethod javaMethod = (JavaMethod)childJavaTerm;
+
+			String javaMethodContent = javaMethod.getContent();
+
+			Matcher upsertCPDefinitionMatcher =
+				_upsertCPDefinitionPattern.matcher(javaMethodContent);
+
+			while (upsertCPDefinitionMatcher.find()) {
+				String methodCall = JavaSourceUtil.getMethodCall(
+					javaMethodContent, upsertCPDefinitionMatcher.start());
+
+				String variableName = upsertCPDefinitionMatcher.group(2);
+
+				if (!hasClassOrVariableName(
+						"CPDefinitionLocalService", content, content,
+						variableName) &&
+					!variableName.contains("CPDefinitionLocalServiceUtil") &&
+					!hasClassOrVariableName(
+						"CPDefinitionService", content, content,
+						variableName) &&
+					!variableName.contains("CPDefinitionServiceUtil")) {
+
+					continue;
+				}
```

Comment:
@TamyrisTorres, the orders shouldn't be like that. If there is only one call and that is a Utill call It will break this logic. Please check the Utils types first.

Reply:
resolved!